topk.go

```diff
@@ -233,84 +265,91 @@ func (s *Stream) Merge(other *Stream) error {
 		return fmt.Errorf("expected stream of size n %d, got %d", s.n, other.n)
 	}
 
-	// merge the elements
-	eKeys := make(map[string]struct{})
-	eMap := make(map[string]Element)
-	for _, e := range s.k.elts {
-		eKeys[e.Key] = struct{}{}
-	}
-	for _, e := range other.k.elts {
-		eKeys[e.Key] = struct{}{}
-	}
+	for i := range s.p {
 
-	for k := range eKeys {
-		idx1, ok1 := s.k.m[k]
-		idx2, ok2 := other.k.m[k]
-		xhash := reduce(metro.Hash64Str(k, 0), len(s.alphas))
-		min1 := other.alphas[xhash]
-		min2 := other.alphas[xhash]
-
-		switch {
-		case ok1 && ok2:
-			e1 := s.k.elts[idx1]
-			e2 := other.k.elts[idx2]
-			eMap[k] = Element{
-				Key:   k,
-				Count: e1.Count + e2.Count,
-				Error: e1.Error + e2.Error,
-			}
-		case ok1:
-			e1 := s.k.elts[idx1]
-			eMap[k] = Element{
-				Key:   k,
-				Count: e1.Count + min2,
-				Error: e1.Error + min2,
-			}
-		case ok2:
-			e2 := other.k.elts[idx2]
-			eMap[k] = Element{
-				Key:   k,
-				Count: e2.Count + min1,
-				Error: e2.Error + min1,
+		// merge the elements
+		eKeys := make(map[string]struct{})
+		eMap := make(map[string]Element)
+		for _, e := range s.p[i].elts {
+			eKeys[e.Key] = struct{}{}
+		}
+		for _, e := range other.p[i].elts {
+			eKeys[e.Key] = struct{}{}
+		}
+
+		for k := range eKeys {
+			idx1, ok1 := s.p[i].m[k]
+			idx2, ok2 := other.p[i].m[k]
+			xhash := reduce(metro.Hash64Str(k, 0), len(s.alphas))
+			min1 := other.alphas[xhash]
+			min2 := other.alphas[xhash]
```

Comment:
min1 and min2 are always assigned the same value, and they appear to only be read inside this scope.  Is there some reason we need both?